### PR TITLE
Add select with cars to route form [fixes #68]

### LINKED ISSUE #68: Drives: car select field
Add car select field to Drive (Route) form.

Preserve cars in local storage (with the middleware we have).
For now we can load cars whenever we open the form. Later we want to load it only when page loads.

### PR DESCRIPTION
**Pull request by juliaanholcer**
[fixes #68]

### CODE REVIEW

File: frontend/src/views/RouteFormView.vue [hunk 1]
```diff
@@ -139,6 +158,12 @@ export default {
         .map(makeErrorMessage(this.$t.bind(this)));
     },
   },
+  created() {
+    this[actions.FETCH_CARS]();
+  },
+  computed: {
+    ...mapState(['cars', 'fetchingCarsInProgress']),
```
Comment (adamgajzlerowicz) [inline frontend/src/views/RouteFormView.vue]: Could those be constants?

File: frontend/src/views/RouteFormView.vue [hunk 2]
```diff
@@ -92,14 +110,15 @@
...
-import { mapActions } from 'vuex';
+import { mapActions, mapState } from 'vuex';
 import uuidv4 from 'uuid/v4';
 import * as actions from '../store/actions';
 import { isErroring, makeErrorMessage } from './services';
 
 const defaultFormState = {
   id: uuidv4(),
   date: '',
+  car: '',
```
Comment (adamgajzlerowicz) [inline frontend/src/views/RouteFormView.vue]: this should be null

File: frontend/src/store/actions.js [hunk 3]
```diff
@@ -14,6 +16,20 @@ export const actions = {
       commit(mutations.SET_USER, user);
     });
   },
+  [FETCH_CARS]({ commit }) {
+    commit(mutations.SET_FETCHING_CARS_PROGRESS, true);
```
Comment (adamgajzlerowicz) [inline frontend/src/store/actions.js]: probably should set errors to null here too

File: frontend/src/store/actions.js [hunk 4]
```diff
@@ -14,6 +16,20 @@ export const actions = {
...
+    getCars()
+      .then((cars) => {
+        commit(mutations.SET_FETCHING_CARS_ERROR, null);
+        commit(mutations.SET_CARS, cars);
+      })
+      .catch(() => {
+        commit(mutations.SET_FETCHING_CARS_ERROR, 'Fetching cars unsuccessful ');
```
Comment (adamgajzlerowicz) [inline frontend/src/store/actions.js]: error is not translated
also loading should be set to false.

File: frontend/src/views/RoutesView.vue [hunk 5]
```diff
@@ -24,15 +24,19 @@
...
+            <span class="font-weight-bold">{{ $t('routes.car') }} </span>
+            <span>{{ route.car.plates }}</span>
+          </p>
+          <p>
+            <span class="font-weight-bold">{{ $t('routes.starting_mileage') }} </span>
             <span>{{ route.startMileage }}</span>
           </p>
           <p>
-            <span class="font-weight-bold">{{ $t('routes.ending_mileage') }}</span>
+            <span class="font-weight-bold">{{ $t('routes.ending_mileage') }} </span>
```
Comment (adamgajzlerowicz) [inline frontend/src/views/RoutesView.vue]: why the extra space? there is no space after span tag..

Comment (juliaanholcer) [reply frontend/src/views/RoutesView.vue]: IMHO it looks a little bit better with space between first and second span. 
zaznaczenie_001

File: frontend/src/store/index.js [hunk 6]
```diff
@@ -17,9 +18,12 @@ export const IS_ONLINE = 'isOnline';
 const state = {
   [USER]: null,
   [ROUTES]: [],
+  [CARS]: [],
   [IS_ONLINE]: navigator.onLine,
   loginInProgress: false,
   loginError: null,
+  fetchingCarsInProgress: false,
```
[the remark this answers is not in the document]
Comment (juliaanholcer) [reply frontend/src/store/index.js]: I agree, I even suggested smth similar two weeks ago, but @jacol12345 wanted to keep store flat. But even now, at the very beginning, it starts to be a little messy.

File: frontend/src/store/mutations.js [hunk 7]
```diff
@@ -1,11 +1,12 @@
-import { setItem } from '../services/localStore';
-
 export const SET_USER = 'SET_USER';
 export const SET_LOGIN_PROGRESS = 'SET_LOGIN_PROGRESS';
 export const SET_LOGIN_ERROR = 'SET_LOGIN_ERROR';
 export const ADD_ROUTE = 'ADD_ROUTE';
 export const SET_UPDATE_READY = 'SET_UPDATE_READY';
 export const SET_IS_CONNECTED = 'SET_IS_CONNECTED';
+export const SET_FETCHING_CARS_ERROR = 'SET_FETCHING_CARS_PROGRESS';
```
[the remark this answers is not in the document]
Comment (juliaanholcer) [reply frontend/src/store/mutations.js]: copy-paste misspell, sorry :flushed:

File: frontend/src/views/RouteFormView.vue (hunk 2 above)
Comment (juliaanholcer) [reply frontend/src/views/RouteFormView.vue]: ok

File: frontend/src/store/actions.js (hunk 3 above)
Comment (juliaanholcer) [reply frontend/src/store/actions.js]: An error is set to null after getCars was resolved successfully. If I understand you correctly I should move this earlier, before fetching data?

**Pull request merged**